src/actions/actions.ts

```diff
@@ -583,6 +589,21 @@ export class DeleteOperatorXVisual extends BaseOperator {
 }
 
 @RegisterAction
+export class UpperCaseOperatorXVisual extends BaseOperator {
+    public keys = ["U"];
+    public modes = [ModeName.Visual, ModeName.VisualLine];
+    
+    public async run(vimState: VimState, start: Position, end: Position): Promise<VimState> {
+      [start, end] = adjustVimSelectionPosition(vimState, start, end);
```

Comment:
Ah; all that logic is contained within executeOperator inside modeHandler.ts.

However, I think that only the swapping of start and stop should be moved in there (that's a good catch, btw!) - the rest of this stuff is DeleteOperator specific.